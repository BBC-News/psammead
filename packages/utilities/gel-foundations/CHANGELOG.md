# Gel Foundations Changelog

| Version | Description |
|---------|-------------|
| 0.1.4   | [PR#224](https://github.com/BBC-News/psammead/pull/224) Add tests for the exported values, coverage 100% :tada: |
| 0.1.1   | [PR#212](https://github.com/BBC-News/psammead/pull/212) Update package description and README. |
| 0.1.0   | [PR#221](https://github.com/BBC-News/psammead/pull/221) Create initial package, pulled in from gel-constants and gel-foundations-styled-components packages. |

