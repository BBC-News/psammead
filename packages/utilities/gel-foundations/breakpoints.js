module.exports = require('./dist/breakpoints'); // eslint-disable-line import/no-unresolved
