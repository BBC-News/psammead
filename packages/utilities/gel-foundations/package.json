{
  "name": "drew-testing-123",
  "version": "1.0.6",
  "main": "dist/index.cjs.js",
  "module": "dist/index.es.js",
  "description": "A range of string constants for use in CSS, intended to help implement BBC GEL-compliant webpages and components.",
  "repository": {
    "type": "git",
    "url": "https://github.com/bbc/psammead/tree/latest/packages/utilities/gel-foundations"
  },
  "author": {
    "name": "Psammead Maintainers",
    "email": "PsammeadMaintainers@bbc.co.uk"
  },
  "files": [
    "dist/"
  ],
  "license": "Apache-2.0",
  "bugs": {
    "url": "https://github.com/bbc/psammead/issues"
  },
  "homepage": "https://github.com/bbc/psammead/blob/latest/packages/utilities/gel-foundations/README.md",
  "keywords": [
    "bbc",
    "utilities",
    "gel",
    "breakpoints",
    "grid",
    "spacing",
    "typography"
  ],
  "devDependencies": {
    "@bbc/psammead-test-helpers": "^1.0.1"
  }
}
