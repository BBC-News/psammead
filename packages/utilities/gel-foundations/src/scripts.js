export latin from './scripts/latin-and-cyrillic';
export cyrillic from './scripts/latin-and-cyrillic';
export latinDiacritics from './scripts/latin-with-diacritics';
export arabic from './scripts/arabic';
export arabicPashto from './scripts/arabic-pashto';
export bengali from './scripts/bengali';
export burmese from './scripts/burmese';
export hindi from './scripts/devanagari-and-gurmukhi';
export nepali from './scripts/devanagari-and-gurmukhi';
export sinhalese from './scripts/sinhalese';
export tamil from './scripts/tamil';
export thai from './scripts/thai';
export chinese from './scripts/no-ascenders-or-descenders';
export korean from './scripts/no-ascenders-or-descenders';
export ethiopic from './scripts/no-ascenders-or-descenders';
