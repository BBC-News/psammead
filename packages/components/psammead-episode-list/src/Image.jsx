import React from 'react';
import { string } from 'prop-types';
import omit from 'ramda/src/omit';
import styled from '@emotion/styled';
import { mediaIcons } from '@bbc/psammead-assets/svgs';
import { getMinion } from '@bbc/gel-foundations/typography';
import { C_EBON, C_WHITE } from '@bbc/psammead-styles/colours';
import { getSansRegular } from '@bbc/psammead-styles/font-styles';
import {
  GEL_SPACING_HLF,
  GEL_SPACING,
  GEL_SPACING_DBL,
} from '@bbc/gel-foundations/spacings';
import {
  GEL_GROUP_2_SCREEN_WIDTH_MIN,
  GEL_GROUP_3_SCREEN_WIDTH_MIN,
} from '@bbc/gel-foundations/breakpoints';

import { withEpisodeLocality } from './helpers';

const Wrapper = styled.div`
  display: inline-block;
  position: relative;
  line-height: 0;
  margin-right: ${GEL_SPACING};
  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
    margin-right: ${GEL_SPACING_DBL};
  }
`;

const PlayWrapper = withEpisodeLocality(styled.div`
  background-color: ${C_EBON};
  padding: ${GEL_SPACING_HLF};
  @media (min-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) {
    padding: ${GEL_SPACING};
  }

  svg {
    margin: 0 0 2px 0;
    height: 10px;
    width: 12px;
    fill: white;
  }

  @media (min-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) {
    position: absolute;
    bottom: 0;
  }
`);

const DurationWrapper = withEpisodeLocality(styled.label`
  ${({ script }) => getMinion(script)};
  ${({ service }) => getSansRegular(service)}
  color: ${C_WHITE};
  padding-left: ${GEL_SPACING_HLF};
`);

const StyledImage = styled.img`
  width: 70px;
  @media (min-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) {
    width: 120px;
  }
  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
    width: 230px;
  }
`;

const EpisodeImage = props => {
  const { duration, alt } = props;

  // This component only uses a subset of its props
  // the remaining props are passed down to the underyling <img> element
  const selectImgProps = omit(['alt', 'duration']);

  return (
    <Wrapper>
      <StyledImage alt={alt} {...selectImgProps(props)} />
      <PlayWrapper>
        {mediaIcons.video}
        {duration && <DurationWrapper>{duration}</DurationWrapper>}
      </PlayWrapper>
    </Wrapper>
  );
};

EpisodeImage.propTypes = {
  alt: string,
  duration: string,
};

EpisodeImage.defaultProps = {
  alt: '',
  duration: '',
};

export default EpisodeImage;
