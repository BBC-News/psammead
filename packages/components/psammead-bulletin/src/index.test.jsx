import React from 'react';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import { latin } from '@bbc/gel-foundations/scripts';
import Image from '@bbc/psammead-image';
import Bulletin from '.';

/* eslint-disable react/prop-types */
const BulletinComponent = ({ script, service, isLive, mediaType, ctaText }) => {
  const summaryText = 'This is the summary text';
  const headlineText = 'This is the headline';
  const ctaLink = 'https://bbc.co.uk';

  const playCtaText = isLive ? `${ctaText} Live` : ctaText;
  const offScreenText = isLive ? `${ctaText} LIVE` : ctaText;

  const image = (
    <Image
      src="https://ichef.bbci.co.uk/news/660/cpsprodpb/11897/production/_106613817_999_al_.jpg"
      alt="Iron man"
    />
  );
  return (
    <Bulletin
      script={script}
      service={service}
      image={image}
      mediaType={mediaType}
      headlineText={headlineText}
      summaryText={summaryText}
      ctaLink={ctaLink}
      ctaText={playCtaText}
      isLive={isLive}
      offScreenText={offScreenText}
    />
  );
};

describe('Bulletin', () => {
  shouldMatchSnapshot(
    'should render audio correctly',
    <BulletinComponent
      script={latin}
      service="news"
      mediaType="audio"
      ctaText="Listen"
    />,
  );

  shouldMatchSnapshot(
    'should render video correctly',
    <BulletinComponent
      script={latin}
      service="news"
      mediaType="video"
      ctaText="Watch"
    />,
  );

  shouldMatchSnapshot(
    'should render live audio correctly',
    <BulletinComponent
      script={latin}
      service="news"
      mediaType="audio"
      ctaText="Listen"
      isLive
    />,
  );

  shouldMatchSnapshot(
    'should render live video correctly',
    <BulletinComponent
      script={latin}
      service="news"
      mediaType="video"
      ctaText="Watch"
      isLive
    />,
  );
});
