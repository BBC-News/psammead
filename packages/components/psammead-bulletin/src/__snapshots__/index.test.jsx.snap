// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bulletin should render audio correctly 1`] = `
.c2 {
  display: block;
  width: 100%;
  visibility: visible;
}

.c10 {
  vertical-align: middle;
  margin: 0 0.25rem;
  color: #222222;
  fill: currentColor;
  width: 0.8125rem;
  height: 0.75rem;
}

.c6 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

.c5 {
  position: static;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5:before {
  bottom: 0;
  content: '';
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  white-space: nowrap;
  z-index: 1;
}

.c5:hover,
.c5:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c5:visited {
  color: #6E6E73;
}

.c1 {
  vertical-align: top;
  display: inline-block;
  width: 100%;
  padding: 0.5rem 0.5rem 0 0.5rem;
}

.c3 {
  display: inline-block;
  width: 100%;
}

.c0 {
  position: relative;
  background-color: #F2F2F2;
  display: grid;
  grid-template-columns: repeat(6,1fr);
  grid-column-gap: 1rem;
  background-color: #F2F2F2;
}

.c4 {
  color: #222222;
  margin: 0;
  padding: 0.5rem;
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.25rem;
}

.c7 {
  color: #3F3F42;
  margin: 0;
  padding: 0 0.5rem 1rem;
  font-size: 0.9375rem;
  line-height: 1.125rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 0.5rem;
}

.c9 > svg {
  color: #FFFFFF;
  fill: currentColor;
  width: 1.0625rem;
  height: 1rem;
  margin: 0;
}

.c8 {
  background-color: #222222;
  border: 0.0625rem solid transparent;
  color: #FFFFFF;
  padding: 0.75rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 0.9375rem;
  line-height: 1.25rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
}

@media (min-width:37.5rem) and (max-width:62.9375rem) {
  .c1 {
    width: 33.33%;
  }
}

@media (min-width:37.5rem) {
  .c1 {
    padding: 0;
  }
}

@supports (grid-template-columns:fit-content(200px)) {
  .c1 {
    width: initial;
    grid-column: 1 / span 6;
  }
}

@media (min-width:37.5rem) and (max-width:62.9375rem) {
  .c3 {
    width: 66.67%;
    padding-left: 1rem;
  }
}

@supports (grid-template-columns:fit-content(200px)) {
  .c3 {
    width: initial;
    grid-column: 1 / span 6;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c4 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) and (max-width:62.9375rem) {
  .c4 {
    padding-top: 0.5rem;
    padding-bottom: 0.5rem;
    padding-left: 0;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c7 {
    font-size: 0.9375rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c7 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) and (max-width:62.9375rem) {
  .c7 {
    padding-left: 0;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c8 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c8 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) and (max-width:62.9375rem) {
  .c8 {
    display: -webkit-inline-box;
    display: -webkit-inline-flex;
    display: -ms-inline-flexbox;
    display: inline-flex;
    padding: 0.5rem 1rem;
    margin-bottom: 1rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <img
      alt="Iron man"
      class="c2"
      src="https://ichef.bbci.co.uk/news/660/cpsprodpb/11897/production/_106613817_999_al_.jpg"
    />
  </div>
  <div
    class="c3"
    dir="ltr"
  >
    <h3
      class="c4"
      dir="ltr"
    >
      <a
        class="c5"
        href="https://bbc.co.uk"
      >
        <span
          role="text"
        >
          <span
            class="c6"
            lang="en-GB"
          >
            Listen, 
          </span>
          <span>
            This is the headline
          </span>
        </span>
      </a>
    </h3>
    <p
      class="c7"
      dir="ltr"
    >
      This is the summary text
    </p>
    <div
      aria-hidden="true"
      class="c8"
      dir="ltr"
    >
      <span
        class="c9"
        dir="ltr"
      >
        <svg
          aria-hidden="true"
          class="c10"
          focusable="false"
          height="12px"
          viewBox="0 0 13 12"
          width="13px"
        >
          <path
            d="M9.021 1.811l-.525.525c.938.938 1.5 2.25 1.5 3.675s-.563 2.738-1.5 3.675l.525.525c1.05-1.087 1.725-2.55 1.725-4.2s-.675-3.112-1.725-4.2z"
          />
          <path
            d="M10.596.199l-.525.562c1.35 1.35 2.175 3.225 2.175 5.25s-.825 3.9-2.175 5.25l.525.525c1.5-1.462 2.4-3.525 2.4-5.775s-.9-4.312-2.4-5.812zM6.996 1.511l-2.25 2.25H.996v4.5h3.75l2.25 2.25z"
          />
        </svg>
      </span>
      Listen
    </div>
  </div>
</div>
`;

exports[`Bulletin should render live audio correctly 1`] = `
.c2 {
  display: block;
  width: 100%;
  visibility: visible;
}

.c11 {
  vertical-align: middle;
  margin: 0 0.25rem;
  color: #222222;
  fill: currentColor;
  width: 0.8125rem;
  height: 0.75rem;
}

.c7 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

.c6 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 700;
  font-style: normal;
  color: #B80000;
  display: inline-block;
  margin-right: 0.5rem;
}

.c5 {
  position: static;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5:before {
  bottom: 0;
  content: '';
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  white-space: nowrap;
  z-index: 1;
}

.c5:hover,
.c5:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c5:visited {
  color: #6E6E73;
}

.c1 {
  vertical-align: top;
  display: inline-block;
  width: 100%;
  padding: 0.5rem 0.5rem 0 0.5rem;
}

.c3 {
  display: inline-block;
  width: 100%;
}

.c0 {
  position: relative;
  background-color: #F2F2F2;
  display: grid;
  grid-template-columns: repeat(6,1fr);
  grid-column-gap: 1rem;
  background-color: #F2F2F2;
}

.c4 {
  color: #222222;
  margin: 0;
  padding: 0.5rem;
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.25rem;
}

.c8 {
  color: #3F3F42;
  margin: 0;
  padding: 0 0.5rem 1rem;
  font-size: 0.9375rem;
  line-height: 1.125rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 0.5rem;
}

.c10 > svg {
  color: #FFFFFF;
  fill: currentColor;
  width: 1.0625rem;
  height: 1rem;
  margin: 0;
}

.c9 {
  background-color: #B80000;
  border: 0.0625rem solid transparent;
  color: #FFFFFF;
  padding: 0.75rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 0.9375rem;
  line-height: 1.25rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
}

@media (min-width:37.5rem) and (max-width:62.9375rem) {
  .c1 {
    width: 33.33%;
  }
}

@media (min-width:37.5rem) {
  .c1 {
    padding: 0;
  }
}

@supports (grid-template-columns:fit-content(200px)) {
  .c1 {
    width: initial;
    grid-column: 1 / span 6;
  }
}

@media (min-width:37.5rem) and (max-width:62.9375rem) {
  .c3 {
    width: 66.67%;
    padding-left: 1rem;
  }
}

@supports (grid-template-columns:fit-content(200px)) {
  .c3 {
    width: initial;
    grid-column: 1 / span 6;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c4 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) and (max-width:62.9375rem) {
  .c4 {
    padding-top: 0.5rem;
    padding-bottom: 0.5rem;
    padding-left: 0;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c8 {
    font-size: 0.9375rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c8 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) and (max-width:62.9375rem) {
  .c8 {
    padding-left: 0;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c9 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c9 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) and (max-width:62.9375rem) {
  .c9 {
    display: -webkit-inline-box;
    display: -webkit-inline-flex;
    display: -ms-inline-flexbox;
    display: inline-flex;
    padding: 0.5rem 1rem;
    margin-bottom: 1rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <img
      alt="Iron man"
      class="c2"
      src="https://ichef.bbci.co.uk/news/660/cpsprodpb/11897/production/_106613817_999_al_.jpg"
    />
  </div>
  <div
    class="c3"
    dir="ltr"
  >
    <h3
      class="c4"
      dir="ltr"
    >
      <a
        class="c5"
        href="https://bbc.co.uk"
      >
        <span
          role="text"
        >
          <span
            aria-hidden="true"
            class="c6"
            dir="ltr"
          >
            LIVE 
          </span>
          <span
            class="c7"
            lang="en-GB"
          >
            Listen LIVE, 
          </span>
          This is the headline
        </span>
      </a>
    </h3>
    <p
      class="c8"
      dir="ltr"
    >
      This is the summary text
    </p>
    <div
      aria-hidden="true"
      class="c9"
      dir="ltr"
    >
      <span
        class="c10"
        dir="ltr"
      >
        <svg
          aria-hidden="true"
          class="c11"
          focusable="false"
          height="12px"
          viewBox="0 0 13 12"
          width="13px"
        >
          <path
            d="M9.021 1.811l-.525.525c.938.938 1.5 2.25 1.5 3.675s-.563 2.738-1.5 3.675l.525.525c1.05-1.087 1.725-2.55 1.725-4.2s-.675-3.112-1.725-4.2z"
          />
          <path
            d="M10.596.199l-.525.562c1.35 1.35 2.175 3.225 2.175 5.25s-.825 3.9-2.175 5.25l.525.525c1.5-1.462 2.4-3.525 2.4-5.775s-.9-4.312-2.4-5.812zM6.996 1.511l-2.25 2.25H.996v4.5h3.75l2.25 2.25z"
          />
        </svg>
      </span>
      Listen Live
    </div>
  </div>
</div>
`;

exports[`Bulletin should render live video correctly 1`] = `
.c2 {
  display: block;
  width: 100%;
  visibility: visible;
}

.c11 {
  vertical-align: middle;
  margin: 0 0.25rem;
  color: #222222;
  fill: currentColor;
  width: 0.75rem;
  height: 0.75rem;
}

.c7 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

.c6 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 700;
  font-style: normal;
  color: #B80000;
  display: inline-block;
  margin-right: 0.5rem;
}

.c5 {
  position: static;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5:before {
  bottom: 0;
  content: '';
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  white-space: nowrap;
  z-index: 1;
}

.c5:hover,
.c5:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c5:visited {
  color: #6E6E73;
}

.c1 {
  vertical-align: top;
  display: inline-block;
  width: 100%;
  padding: 0.5rem 0.5rem 0 0.5rem;
}

.c3 {
  display: inline-block;
  width: 100%;
}

.c0 {
  position: relative;
  background-color: #F2F2F2;
  display: grid;
  grid-template-columns: repeat(6,1fr);
  grid-column-gap: 1rem;
}

.c4 {
  color: #222222;
  margin: 0;
  padding: 0.5rem;
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-style: normal;
  font-size: 1.125rem;
  line-height: 1.375rem;
}

.c8 {
  color: #3F3F42;
  margin: 0;
  padding: 0 0.5rem 1rem;
  font-size: 0.9375rem;
  line-height: 1.125rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 0.5rem;
}

.c10 > svg {
  color: #FFFFFF;
  fill: currentColor;
  width: 1.0625rem;
  height: 1rem;
  margin: 0;
}

.c9 {
  background-color: #B80000;
  border: 0.0625rem solid transparent;
  color: #FFFFFF;
  padding: 0.75rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 0.9375rem;
  line-height: 1.25rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
}

@media (min-width:37.5rem) {
  .c1 {
    width: 50%;
  }
}

@media (min-width:37.5rem) {
  .c1 {
    padding: 0;
  }
}

@supports (grid-template-columns:fit-content(200px)) {
  .c1 {
    width: initial;
    grid-column: 1 / span 6;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    width: 50%;
    padding-left: 1rem;
  }
}

@supports (grid-template-columns:fit-content(200px)) {
  .c3 {
    width: initial;
    grid-column: 1 / span 6;
  }
}

@media (min-width:37.5rem) {
  .c0 {
    padding: 1rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c4 {
    font-size: 1.125rem;
    line-height: 1.375rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    font-size: 1.25rem;
    line-height: 1.5rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    padding: 0 0 0.5rem 0;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c8 {
    font-size: 0.9375rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c8 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c8 {
    padding-left: 0;
    padding-right: 0;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c9 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c9 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c9 {
    display: -webkit-inline-box;
    display: -webkit-inline-flex;
    display: -ms-inline-flexbox;
    display: inline-flex;
    padding: 0.5rem 1rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <img
      alt="Iron man"
      class="c2"
      src="https://ichef.bbci.co.uk/news/660/cpsprodpb/11897/production/_106613817_999_al_.jpg"
    />
  </div>
  <div
    class="c3"
    dir="ltr"
  >
    <h3
      class="c4"
      dir="ltr"
    >
      <a
        class="c5"
        href="https://bbc.co.uk"
      >
        <span
          role="text"
        >
          <span
            aria-hidden="true"
            class="c6"
            dir="ltr"
          >
            LIVE 
          </span>
          <span
            class="c7"
            lang="en-GB"
          >
            Watch LIVE, 
          </span>
          This is the headline
        </span>
      </a>
    </h3>
    <p
      class="c8"
      dir="ltr"
    >
      This is the summary text
    </p>
    <div
      aria-hidden="true"
      class="c9"
      dir="ltr"
    >
      <span
        class="c10"
        dir="ltr"
      >
        <svg
          aria-hidden="true"
          class="c11"
          focusable="false"
          height="12px"
          viewBox="0 0 32 32"
          width="12px"
        >
          <polygon
            points="3,32 29,16 3,0"
          />
        </svg>
      </span>
      Watch Live
    </div>
  </div>
</div>
`;

exports[`Bulletin should render video correctly 1`] = `
.c2 {
  display: block;
  width: 100%;
  visibility: visible;
}

.c10 {
  vertical-align: middle;
  margin: 0 0.25rem;
  color: #222222;
  fill: currentColor;
  width: 0.75rem;
  height: 0.75rem;
}

.c6 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

.c5 {
  position: static;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5:before {
  bottom: 0;
  content: '';
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  white-space: nowrap;
  z-index: 1;
}

.c5:hover,
.c5:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c5:visited {
  color: #6E6E73;
}

.c1 {
  vertical-align: top;
  display: inline-block;
  width: 100%;
  padding: 0.5rem 0.5rem 0 0.5rem;
}

.c3 {
  display: inline-block;
  width: 100%;
}

.c0 {
  position: relative;
  background-color: #F2F2F2;
  display: grid;
  grid-template-columns: repeat(6,1fr);
  grid-column-gap: 1rem;
}

.c4 {
  color: #222222;
  margin: 0;
  padding: 0.5rem;
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-style: normal;
  font-size: 1.125rem;
  line-height: 1.375rem;
}

.c7 {
  color: #3F3F42;
  margin: 0;
  padding: 0 0.5rem 1rem;
  font-size: 0.9375rem;
  line-height: 1.125rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 0.5rem;
}

.c9 > svg {
  color: #FFFFFF;
  fill: currentColor;
  width: 1.0625rem;
  height: 1rem;
  margin: 0;
}

.c8 {
  background-color: #222222;
  border: 0.0625rem solid transparent;
  color: #FFFFFF;
  padding: 0.75rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 0.9375rem;
  line-height: 1.25rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
}

@media (min-width:37.5rem) {
  .c1 {
    width: 50%;
  }
}

@media (min-width:37.5rem) {
  .c1 {
    padding: 0;
  }
}

@supports (grid-template-columns:fit-content(200px)) {
  .c1 {
    width: initial;
    grid-column: 1 / span 6;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    width: 50%;
    padding-left: 1rem;
  }
}

@supports (grid-template-columns:fit-content(200px)) {
  .c3 {
    width: initial;
    grid-column: 1 / span 6;
  }
}

@media (min-width:37.5rem) {
  .c0 {
    padding: 1rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c4 {
    font-size: 1.125rem;
    line-height: 1.375rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    font-size: 1.25rem;
    line-height: 1.5rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    padding: 0 0 0.5rem 0;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c7 {
    font-size: 0.9375rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c7 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c7 {
    padding-left: 0;
    padding-right: 0;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c8 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c8 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c8 {
    display: -webkit-inline-box;
    display: -webkit-inline-flex;
    display: -ms-inline-flexbox;
    display: inline-flex;
    padding: 0.5rem 1rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <img
      alt="Iron man"
      class="c2"
      src="https://ichef.bbci.co.uk/news/660/cpsprodpb/11897/production/_106613817_999_al_.jpg"
    />
  </div>
  <div
    class="c3"
    dir="ltr"
  >
    <h3
      class="c4"
      dir="ltr"
    >
      <a
        class="c5"
        href="https://bbc.co.uk"
      >
        <span
          role="text"
        >
          <span
            class="c6"
            lang="en-GB"
          >
            Watch, 
          </span>
          <span>
            This is the headline
          </span>
        </span>
      </a>
    </h3>
    <p
      class="c7"
      dir="ltr"
    >
      This is the summary text
    </p>
    <div
      aria-hidden="true"
      class="c8"
      dir="ltr"
    >
      <span
        class="c9"
        dir="ltr"
      >
        <svg
          aria-hidden="true"
          class="c10"
          focusable="false"
          height="12px"
          viewBox="0 0 32 32"
          width="12px"
        >
          <polygon
            points="3,32 29,16 3,0"
          />
        </svg>
      </span>
      Watch
    </div>
  </div>
</div>
`;
